ui-lib/src/androidTest/java/co/electriccoin/zcash/ui/screen/send/integration/SendViewIntegrationTest.kt

```diff
@@ -0,0 +1,69 @@
+package co.electriccoin.zcash.ui.screen.send.integration
+
+import androidx.compose.ui.test.junit4.StateRestorationTester
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.test.filters.MediumTest
+import cash.z.ecc.android.sdk.model.ZecSend
+import cash.z.ecc.sdk.fixture.ZecSendFixture
+import co.electriccoin.zcash.ui.R
+import co.electriccoin.zcash.ui.screen.send.SendViewTestSetup
+import co.electriccoin.zcash.ui.screen.send.model.SendStage
+import co.electriccoin.zcash.ui.test.getStringResource
+import kotlinx.coroutines.runBlocking
+import org.junit.Rule
+import org.junit.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+
+class SendViewIntegrationTest {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    private fun newTestSetup(
+        sendStage: SendStage = SendStage.Form,
+        zecSend: ZecSend? = null
+    ) = SendViewTestSetup(
+        composeTestRule,
+        sendStage,
+        zecSend
+    )
+
+    @Test
+    @MediumTest
+    fun send_screens_values_state_restoration() {
```

Comment:
Is this testing what you want it to test?

If using the `SendViewTestSetup`, we are testing the state restoration logic in SendViewTestSetup's Composable.  I think you want to test the state restoration inside `AndroidSend.WrapSend()` which might require adding one more layer of indirection with 

```
@Composable
private fun WrapSend(
    activity: ComponentActivity,
    goBack: () -> Unit
) {
    val walletViewModel by activity.viewModels<WalletViewModel>()

    val synchronizer = walletViewModel.synchronizer.collectAsStateWithLifecycle().value

    val spendableBalance = walletViewModel.walletSnapshot.collectAsStateWithLifecycle().value?.spendableBalance()

    val spendingKey = walletViewModel.spendingKey.collectAsStateWithLifecycle().value

    WrapSend(synchronizer, spendableBalance, spendingKey, goBack)
}

@VisibleForTesting
@Composable
internal fun WrapSend(
  synchronizer,
  spendableBalance,
  spendingKey,
  goBack
) {
    val scope = rememberCoroutineScope()

    // For now, we're avoiding sub-navigation to keep the navigation logic simple.  But this might
    // change once deep-linking support  is added.  It depends on whether deep linking should do one of:
    // 1. Use a different UI flow entirely
    // 2. Show a pre-filled Send form
    // 3. Go directly to the Confirmation screen
    val (sendStage, setSendStage) = rememberSaveable { mutableStateOf(SendStage.Form) }

    val (zecSend, setZecSend) = rememberSaveable(stateSaver = ZecSend.Saver) { mutableStateOf(null) }

    ... And the rest of the existing WrapSend method
}
```